nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,466 @@
+                    for (String recordPath : columnsToHash) {
+                        RecordPathResult evaluate = RecordPath.compile("hash(" + recordPath + ", '" + hashingAlgorithm + "')").evaluate(currentRecord);
+                        evaluate.getSelectedFields().forEach(fieldVal -> fieldVal.updateValue(fieldVal.getValue(), RecordFieldType.STRING.getDataType()));
+                    }
+                    currentRecord.incorporateInactiveFields();
+                    recordSetWriter.write(currentRecord);
+                    numberOfRecords++;
+                }
+            }
+        }
+        return new TransformResult(numberOfRecords, mimeType);
+    }
+
+    private void route(FlowFile request, FlowFile response, ProcessSession session, ProcessContext context, int statusCode) {
+        if (!isSuccess(statusCode) && request == null) {
+            context.yield();
+        }
+
+        if (isSuccess(statusCode)) {
+            if (request != null) {
+                session.transfer(request, ORIGINAL);
+            }
+            if (response != null) {
+                session.transfer(response, SUCCESS);
+            }
+        } else {
+            if (request != null) {
+                session.transfer(request, FAILURE);
+            }
+        }
+    }
+
+    private boolean isSuccess(int statusCode) {
+        return statusCode >= 200 && statusCode < 300;
+    }
+
+    private Map<String, String> createCommonAttributes(URI uri, HttpResponseEntity httpResponseEntity, long elapsedTime) {
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(STATUS_CODE, String.valueOf(httpResponseEntity.statusCode()));
+        attributes.put(REQUEST_URL, uri.toString());
+        attributes.put(REQUEST_DURATION, Long.toString(elapsedTime));
+        attributes.put(TRANSACTION_ID, UUID.randomUUID().toString());
+        return attributes;
+    }
+
+    private Map<String, String> setMimeType(Map<String, String> commonAttributes, HttpResponseEntity httpResponseEntity) {
+        Map<String, String> attributes = commonAttributes;
+        Optional<String> contentType = httpResponseEntity.headers().getFirstHeader(HEADER_CONTENT_TYPE);
+        if (contentType.isPresent()) {
+            attributes = new HashMap<>(commonAttributes);
+            attributes.put(CoreAttributes.MIME_TYPE.key(), contentType.get());
+        }
+        return attributes;
+    }
+
+    protected enum HashAlgorithm {
+        SHA256("SHA-256"),
+        SHA512("SHA-512");
+
+        private final String name;
+
+        HashAlgorithm(String name) {
+            this.name = name;
+        }
+
+        private String getName() {
+            return name;
+        }
+
+        private static String[] getNames() {
+            return Arrays.stream(HashAlgorithm.values()).map(HashAlgorithm::getName).toArray(String[]::new);
+        }
+    }
+
+    private class TransformResult {
```

Comment:
This class could be declared `static` since it does not reference any instance variables of the parent class.

Reply:
done